src/main/scala/com/mwz/sonar/scala/pr/GlobalIssues.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2018-2019  All sonar-scala contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mwz.sonar.scala
+package pr
+
+import org.sonar.api.batch.fs.InputFile
+import org.sonar.api.batch.{InstantiationStrategy, ScannerSide}
+
+import scala.collection.mutable
+
+/**
+ * Global collection of issues reported during project analysis.
+ * Instantiated only once for a batch run.
+ */
+// TODO: Both @ScannerSide and @InstantiationStrategy are deprecated, we should switch
+//  to the org.sonar.api.scanner.ScannerSide in the future.
+@ScannerSide
+@InstantiationStrategy(InstantiationStrategy.PER_BATCH)
+final class GlobalIssues {
+  // As far as I know this won't be accessed concurrently.
```

Comment:
I am not 100% sure that is a fact.
It is true that the analysis seems serial, but that is outside of our control.
Maybe, someday, sonar decides to run the analysis on parallel. Also, since this is, probably, responsibility of the scanner. Maybe there are custom scanners out there that already do that, who knows.

Since this is outside of our domain, IMHO is better to play safe and, either, use synchronization or _(even better?)_ a **Java** [`ConcurrentHashMap`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html).

Reply:
That's a fair point, I'll use Scala's concurrent TrieMap instead.

Reply:
I had the impression, I had read somewhere that the **Scala's** concurrent collection were totally broken, and that it was better to use the **Java** ones.
However, since most of the time I had immutable collections, I really never need that. And I can't find any proper source for supporting that statement.
I think it would be good to investigate about that.

Reply:
There used to be some issues with the `TrieMap` in the past, but I think they fixed those. Anyway, I couldn't find anything as nice as the `merge` function from the `ConcurrentHashMap`, so I went with that.